Complexity. Displayed below the top block (similar to block comments)

sdh11 opened the pull request:
Added the complexity measure for a flowgraph. Value is now shown below the top block in a similar fashion to block comments. Can be enabled and disabled from the Tools menu.

skoslowski commented on grc/gui/ActionHandler.py:
```diff
@@ -127,6 +127,7 @@ def _handle_action(self, action):
                 Actions.TOGGLE_AUTO_HIDE_PORT_LABELS, Actions.TOGGLE_SNAP_TO_GRID,
                 Actions.TOGGLE_SHOW_BLOCK_COMMENTS,
                 Actions.TOGGLE_SHOW_CODE_PREVIEW_TAB,
+                Actions.TOGGLE_SHOW_FLOWGRAPH_COMPLEXITY,
```
You'll have to add this to the list of actions to load from prefs. at line 146. Better yet we can get rid of that for loop there and add ``` if hasattr(action, 'load_from_preferences'): action.load_from_preferences() ``` to the loop above.

skoslowski commented on grc/gui/Block.py:
```diff
@@ -39,6 +40,15 @@
+<span foreground="#444" size="medium" font_desc="$font"><b>$encode("Complexity: {}bal".format($complexity))</b></span>"""
+COMMENT_COMPLEXITY_MARKUP_TMPL="""\
+#set $foreground = $block.get_enabled() and '#444' or '#888'
+<span foreground="#444" size="medium" font_desc="$font"><b>$encode("Complexity: {}bal".format($complexity))</b></span>
```
A template within a template =)
Use one template. Pass ```complexity=None``` to hide it and you can get rid of that logic in create_comment_label() altogether.

skoslowski commented on grc/gui/Block.py:
```diff
@@ -273,9 +304,10 @@ def what_is_selected(self, coor, coor_m=None):
+        if self._comment_pixmap:
+            window.draw_drawable(gc, self._comment_pixmap, 0, 0, x,
+                                 y + self.H + BLOCK_LABEL_PADDING, -1, -1)
+            comment_height = self._comment_pixmap.get_size()[1] + 5
```
What do you need the comment_height for?

sdh11 replied:
Whoops. Forgot to remove that.

skoslowski commented on grc/gui/Constants.py:
```diff
@@ -59,7 +59,7 @@
 #  flow-graph canvas fonts
 try:  # ugly, but matches current code style
-    FONT_SIZE = _gr_prefs.get_long('grc', 'canvas_font_size', 8)
+    FONT_SIZE = _gr_prefs.get_long('grc', 'canvas_font_size', 10)
```
Why change that. (The prefs have 8 set by default anyway)

sdh11 replied:
On OS X, a font size of 8 is really small and difficult to read. I need to make this part of a separate PR anyway.

skoslowski commented on grc/python/Param.py:
```diff
@@ -88,18 +94,7 @@ def _truncate(string, style=0):
-                elif num.real == 0: return '%sj'%eng_notation.num_to_str(num.imag) #value is imaginary
-                elif num.imag < 0: return '%s-%sj'%(eng_notation.num_to_str(num.real), eng_notation.num_to_str(abs(num.imag)))
-                else: return '%s+%sj'%(eng_notation.num_to_str(num.real), eng_notation.num_to_str(num.imag))
-            else: return str(num)
```
I kind excepted some usages of this function to change? Aren't there any?

sdh11 replied:
I don't believe so. Since it was nested, the __repr__() function was the only one to ever use it. Since I just moved it to a global declaration in that file, the usage syntax didn't need to be changed.

skoslowski commented on grc/gui/Utils.py:
```diff
@@ -21,6 +21,8 @@
 import gtk
 import gobject
+import numpy
+from gnuradio import eng_notation
```
?

skoslowski commented:
Please always double-check the complete diff of your PR (against its base) after making changes. This way you easily spot obsolete changes that remain after editing the original PR with additional commits.

sdh11 commented:
Ok. I addressed everything you mentioned, cleaned up the old changes, rebased to master_grcwg, and squashed the commits. I checked over this diff and I think it looks good.

skoslowski commented:
Great. Thanks. I think this is about ready to be merged. I had three minor changes which I amended your commit. I also added some padding around the comment boxes. See

https://github.com/skoslowski/gnuradio-wg-grc/commits/complexity

I will merge later today if there are no objections

sdh11 commented:
Looks good to me.
